{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e01b7c54",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/liulab/galib/mm_dlbcl_manuscript/scripts'"
      ],
      "text/latex": [
       "'/liulab/galib/mm\\_dlbcl\\_manuscript/scripts'"
      ],
      "text/markdown": [
       "'/liulab/galib/mm_dlbcl_manuscript/scripts'"
      ],
      "text/plain": [
       "[1] \"/liulab/galib/mm_dlbcl_manuscript/scripts\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘tidyverse’ was built under R version 4.1.3”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.6      \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.8      \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.10\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0      \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.1 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2      \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1 \n",
      "\n",
      "Warning message:\n",
      "“package ‘tidyr’ was built under R version 4.1.2”\n",
      "Warning message:\n",
      "“package ‘readr’ was built under R version 4.1.2”\n",
      "Warning message:\n",
      "“package ‘forcats’ was built under R version 4.1.3”\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "Attaching sp\n",
      "\n",
      "Warning message:\n",
      "“package ‘ComplexHeatmap’ was built under R version 4.1.3”\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.10.0\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "  genomic data. Bioinformatics 2016.\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘circlize’ was built under R version 4.1.3”\n",
      "========================================\n",
      "circlize version 0.4.15\n",
      "CRAN page: https://cran.r-project.org/package=circlize\n",
      "Github page: https://github.com/jokergoo/circlize\n",
      "Documentation: https://jokergoo.github.io/circlize_book/book/\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. circlize implements and enhances circular visualization\n",
      "  in R. Bioinformatics 2014.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(circlize))\n",
      "========================================\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘Polychrome’ was built under R version 4.1.3”\n",
      "Warning message:\n",
      "“package ‘RColorBrewer’ was built under R version 4.1.3”\n"
     ]
    }
   ],
   "source": [
    "getwd()\n",
    "setwd(\"/liulab/galib/mm_dlbcl_manuscript/\")\n",
    "library(tidyverse)\n",
    "library(Seurat)\n",
    "library(ktplots)\n",
    "library(ComplexHeatmap)\n",
    "library(circlize)\n",
    "library(Polychrome)\n",
    "library(RColorBrewer)\n",
    "library(pals)\n",
    "source('./scripts/scplot.R')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "36a80214",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read bcl6 single\n",
    "ph_bcl6_6mos_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_6mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_6mos_means <- read.delim(\"./data/cellphoneDB/output/bcl6_6mos_out/means.txt\", check.names = FALSE)\n",
    "ph_bcl6_14mos_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_14mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_14mos_means <- read.delim(\"./data/cellphoneDB/output/bcl6_14mos_out/means.txt\", check.names = FALSE)\n",
    "ph_bcl6_18mos_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_18mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_18mos_means <- read.delim(\"./data/cellphoneDB/output/bcl6_18mos_out/means.txt\", check.names = FALSE)\n",
    "ph_bcl6_sick_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_sick_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_sick_means <- read.delim(\"./data/cellphoneDB/output/bcl6_sick_out/means.txt\", check.names = FALSE)\n",
    "\n",
    "# read bcl6 cd70 double\n",
    "ph_bcl6_double_6mos_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_double_6mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_6mos_means <- read.delim(\"./data/cellphoneDB/output/bcl6_double_6mos_out/means.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_14mos_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_double_14mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_14mos_means <- read.delim(\"./data/cellphoneDB/output/bcl6_double_14mos_out/means.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_18mos_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_double_18mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_18mos_means <- read.delim(\"./data/cellphoneDB/output/bcl6_double_18mos_out/means.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_sick_pvals <- read.delim(\"./data/cellphoneDB/output/bcl6_double_sick_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_bcl6_double_sick_means <- read.delim(\"./data/cellphoneDB/output/bcl6_double_sick_out/means.txt\", check.names = FALSE)\n",
    "\n",
    "# read wt\n",
    "ph_wt_6mos_pvals <- read.delim(\"./data/cellphoneDB/output/wt_6mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_wt_6mos_means <- read.delim(\"./data/cellphoneDB/output/wt_6mos_out/means.txt\", check.names = FALSE)\n",
    "ph_wt_14mos_pvals <- read.delim(\"./data/cellphoneDB/output/wt_14mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_wt_14mos_means <- read.delim(\"./data/cellphoneDB/output/wt_14mos_out/means.txt\", check.names = FALSE)\n",
    "ph_wt_18mos_pvals <- read.delim(\"./data/cellphoneDB/output/wt_18mos_out/pvalues.txt\", check.names = FALSE)\n",
    "ph_wt_18mos_means <- read.delim(\"./data/cellphoneDB/output/wt_18mos_out/means.txt\", check.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "16e22d60",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd3_pos_cd8_neg<- readRDS(\"./data/objects/cd3_pos_cd8_neg_final.rds\")\n",
    "B_cell<- readRDS(\"./data/objects/B_cell_final.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "103f9ef9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# CD4 CTLs (C1, C4a, C6, C13) vs B cells in clusters: \n",
    "# Aged/autoimmune (C7, C8)\n",
    "# Innate-like (C4, C5)\n",
    "# MHC-II-low Cycling (C18, C19, C21)\n",
    "\n",
    "\n",
    "# For WT, Bcl6 and Double at 6, 14 and 18 mos + sick."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12cf59e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd3_pos_cd8_neg_CTLs_clusters<- c(\"cd3_pos_cd8_neg_CTLs\")\n",
    "\n",
    "B_cell_clusters<- c(\"B_cell_Innate-like\", \"B_cell_Aged/autoimmune\", \"B_MHC-II-low Cycling\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ee2a250",
   "metadata": {},
   "source": [
    "## cd3_pos_cd8_neg_CTLs vs B_cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a70e3de4",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd3_pos_cd8_neg_CTLs_B_cell_interacts<- c()\n",
    "B_cell_cd3_pos_cd8_neg_CTLs_interacts<- c()\n",
    "\n",
    "for (c1 in cd3_pos_cd8_neg_CTLs_clusters){\n",
    "    for (c2 in B_cell_clusters){\n",
    "        \n",
    "        interact<- paste0(c1, \"|\", c2)\n",
    "        inverse_interact<- paste0(c2, \"|\", c1)\n",
    "        \n",
    "        cd3_pos_cd8_neg_CTLs_B_cell_interacts<- c(cd3_pos_cd8_neg_CTLs_B_cell_interacts, interact)\n",
    "        B_cell_cd3_pos_cd8_neg_CTLs_interacts<- c(B_cell_cd3_pos_cd8_neg_CTLs_interacts, inverse_interact)\n",
    "    }\n",
    "}\n",
    "\n",
    "cd3_pos_cd8_neg_CTLs_B_cell_interacts\n",
    "sum(colnames(ph_wt_6mos_pvals) %in% cd3_pos_cd8_neg_CTLs_B_cell_interacts) == length(cd3_pos_cd8_neg_CTLs_B_cell_interacts)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_CTLs_interacts\n",
    "sum(colnames(ph_wt_6mos_pvals) %in% B_cell_cd3_pos_cd8_neg_CTLs_interacts) == length(B_cell_cd3_pos_cd8_neg_CTLs_interacts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0edaa9d2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "total number of intersections considered: 26\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 23\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 20\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 20\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 21\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 19\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cd3_pos_cd8_neg_B_cell_mtx_bcl6 = build_cpdb_mtx(cols = cd3_pos_cd8_neg_CTLs_B_cell_interacts,\n",
    "                                            scaled = FALSE,\n",
    "                                            sig.cellpairs = 3,\n",
    "                                            sig.timepoint = 1,  \n",
    "                                            group_name = \"_bcl6\",\n",
    "                                            ph_6mos_pvals = ph_bcl6_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_sick_means)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_mtx_bcl6 = build_cpdb_mtx(cols = B_cell_cd3_pos_cd8_neg_CTLs_interacts, \n",
    "                                            scaled = FALSE,\n",
    "                                            sig.cellpairs = 3,\n",
    "                                            sig.timepoint = 1,  \n",
    "                                            group_name = \"_bcl6\",\n",
    "                                            ph_6mos_pvals = ph_bcl6_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_sick_means)\n",
    "\n",
    "cd3_pos_cd8_neg_B_cell_mtx_bcl6_double = build_cpdb_mtx(cols = cd3_pos_cd8_neg_CTLs_B_cell_interacts,\n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_bcl6_double\",\n",
    "                                            sig.cellpairs = 3,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_bcl6_double_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_double_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_double_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_double_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_double_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_double_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_double_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_double_sick_means)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_mtx_bcl6_double = build_cpdb_mtx(cols = B_cell_cd3_pos_cd8_neg_CTLs_interacts, \n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_bcl6_double\",\n",
    "                                            sig.cellpairs = 3,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_bcl6_double_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_double_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_double_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_double_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_double_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_double_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_double_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_double_sick_means)\n",
    "\n",
    "cd3_pos_cd8_neg_B_cell_mtx_wt = build_cpdb_mtx_wt(cols = cd3_pos_cd8_neg_CTLs_B_cell_interacts,\n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_wt\",\n",
    "                                            sig.cellpairs = 3,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_wt_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_wt_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_wt_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_wt_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_wt_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_wt_18mos_means)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_mtx_wt = build_cpdb_mtx_wt(cols = B_cell_cd3_pos_cd8_neg_CTLs_interacts, \n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_wt\",\n",
    "                                            sig.cellpairs = 3,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_wt_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_wt_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_wt_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_wt_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_wt_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_wt_18mos_means)\n",
    "\n",
    "\n",
    "cd3_pos_cd8_neg_CTLs_B_cell_mtx<- merge_mtx(cd3_pos_cd8_neg_B_cell_mtx_wt, cd3_pos_cd8_neg_B_cell_mtx_bcl6, cd3_pos_cd8_neg_B_cell_mtx_bcl6_double)\n",
    "B_cell_cd3_pos_cd8_neg_CTLs_mtx<- merge_mtx(B_cell_cd3_pos_cd8_neg_mtx_wt, B_cell_cd3_pos_cd8_neg_mtx_bcl6, B_cell_cd3_pos_cd8_neg_mtx_bcl6_double)\n",
    "\n",
    "hp1 = plt_cpdb_dotplot_with_sick(exp_mtx = cd3_pos_cd8_neg_CTLs_B_cell_mtx$exp_mtx, \n",
    "                       pval_mtx = cd3_pos_cd8_neg_CTLs_B_cell_mtx$pval_mtx)\n",
    "hp2 = plt_cpdb_dotplot_with_sick(exp_mtx = B_cell_cd3_pos_cd8_neg_CTLs_mtx$exp_mtx, \n",
    "                       pval_mtx = B_cell_cd3_pos_cd8_neg_CTLs_mtx$pval_mtx)\n",
    "\n",
    "pdf(\"./results/figures/9_Dotplot_cd3_pos_cd8_neg_CTLs_B_cell_wt_bcl6_and_bcl6_double_cell3_time1_genotype1.pdf\", width = 12, height = 10)\n",
    "draw(hp1, annotation_legend_list = lgd, ht_gap = unit(1, \"cm\"))\n",
    "dev.off()\n",
    "\n",
    "pdf(\"./results/figures/9_Dotplot_B_cell_cd3_pos_cd8_neg_CTLs_wt_bcl6_and_bcl6_double_cell3_time1_genotype1.pdf\", width = 12, height = 10)\n",
    "draw(hp2, annotation_legend_list = lgd, ht_gap = unit(1, \"cm\"))\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "af95e5b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "total number of intersections considered: 40\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 48\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 40\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 45\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "sick...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 44\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "total number of intersections considered: 38\n",
      "\n",
      "6 mos...\n",
      "\n",
      "14 mos...\n",
      "\n",
      "18 mos...\n",
      "\n",
      "Generating count matrix...\n",
      "\n",
      "Generating pvalue matrix...\n",
      "\n",
      "output unscaled data\n",
      "\n",
      "wt...\n",
      "\n",
      "bcl6...\n",
      "\n",
      "bcl6 double...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] NA NA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TRUE\n",
      "\n",
      "scaling the count matrix\n",
      "\n",
      "wt...\n",
      "\n",
      "bcl6...\n",
      "\n",
      "bcl6 double...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] NA NA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TRUE\n",
      "\n",
      "scaling the count matrix\n",
      "\n",
      "quantile: -1.1087756153354-0.3253213656421291.91819397336388\n",
      "\n",
      "12129\n",
      "\n",
      "9996\n",
      "\n",
      "quantile: -1.16205336670373-0.183710169412131.79918673420415\n",
      "\n",
      "12129\n",
      "\n",
      "9996\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cd3_pos_cd8_neg_B_cell_mtx_bcl6 = build_cpdb_mtx(cols = cd3_pos_cd8_neg_CTLs_B_cell_interacts,\n",
    "                                            scaled = FALSE,\n",
    "                                            sig.cellpairs = 1,\n",
    "                                            sig.timepoint = 1,  \n",
    "                                            group_name = \"_bcl6\",\n",
    "                                            ph_6mos_pvals = ph_bcl6_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_sick_means)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_mtx_bcl6 = build_cpdb_mtx(cols = B_cell_cd3_pos_cd8_neg_CTLs_interacts, \n",
    "                                            scaled = FALSE,\n",
    "                                            sig.cellpairs = 1,\n",
    "                                            sig.timepoint = 1,  \n",
    "                                            group_name = \"_bcl6\",\n",
    "                                            ph_6mos_pvals = ph_bcl6_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_sick_means)\n",
    "\n",
    "cd3_pos_cd8_neg_B_cell_mtx_bcl6_double = build_cpdb_mtx(cols = cd3_pos_cd8_neg_CTLs_B_cell_interacts,\n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_bcl6_double\",\n",
    "                                            sig.cellpairs = 1,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_bcl6_double_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_double_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_double_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_double_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_double_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_double_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_double_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_double_sick_means)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_mtx_bcl6_double = build_cpdb_mtx(cols = B_cell_cd3_pos_cd8_neg_CTLs_interacts, \n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_bcl6_double\",\n",
    "                                            sig.cellpairs = 1,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_bcl6_double_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_bcl6_double_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_bcl6_double_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_bcl6_double_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_bcl6_double_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_bcl6_double_18mos_means,\n",
    "                                            ph_sick_pvals = ph_bcl6_double_sick_pvals,\n",
    "                                            ph_sick_means = ph_bcl6_double_sick_means)\n",
    "\n",
    "cd3_pos_cd8_neg_B_cell_mtx_wt = build_cpdb_mtx_wt(cols = cd3_pos_cd8_neg_CTLs_B_cell_interacts,\n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_wt\",\n",
    "                                            sig.cellpairs = 1,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_wt_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_wt_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_wt_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_wt_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_wt_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_wt_18mos_means)\n",
    "\n",
    "B_cell_cd3_pos_cd8_neg_mtx_wt = build_cpdb_mtx_wt(cols = B_cell_cd3_pos_cd8_neg_CTLs_interacts, \n",
    "                                            scaled = FALSE,\n",
    "                                            group_name = \"_wt\",\n",
    "                                            sig.cellpairs = 1,\n",
    "                                            sig.timepoint = 1, \n",
    "                                            ph_6mos_pvals = ph_wt_6mos_pvals,\n",
    "                                            ph_6mos_means = ph_wt_6mos_means,\n",
    "                                            ph_14mos_pvals = ph_wt_14mos_pvals,\n",
    "                                            ph_14mos_means = ph_wt_14mos_means,\n",
    "                                            ph_18mos_pvals = ph_wt_18mos_pvals,\n",
    "                                            ph_18mos_means = ph_wt_18mos_means)\n",
    "\n",
    "\n",
    "cd3_pos_cd8_neg_CTLs_B_cell_mtx<- merge_mtx(cd3_pos_cd8_neg_B_cell_mtx_wt, cd3_pos_cd8_neg_B_cell_mtx_bcl6, cd3_pos_cd8_neg_B_cell_mtx_bcl6_double)\n",
    "B_cell_cd3_pos_cd8_neg_CTLs_mtx<- merge_mtx(B_cell_cd3_pos_cd8_neg_mtx_wt, B_cell_cd3_pos_cd8_neg_mtx_bcl6, B_cell_cd3_pos_cd8_neg_mtx_bcl6_double)\n",
    "\n",
    "hp1 = plt_cpdb_dotplot_with_sick(exp_mtx = cd3_pos_cd8_neg_CTLs_B_cell_mtx$exp_mtx, \n",
    "                       pval_mtx = cd3_pos_cd8_neg_CTLs_B_cell_mtx$pval_mtx)\n",
    "hp2 = plt_cpdb_dotplot_with_sick(exp_mtx = B_cell_cd3_pos_cd8_neg_CTLs_mtx$exp_mtx, \n",
    "                       pval_mtx = B_cell_cd3_pos_cd8_neg_CTLs_mtx$pval_mtx)\n",
    "\n",
    "pdf(\"./results/figures/9_Dotplot_cd3_pos_cd8_neg_CTLs_B_cell_wt_bcl6_and_bcl6_double_cell1_time1_genotype1.pdf\", width = 12, height = 10)\n",
    "draw(hp1, annotation_legend_list = lgd, ht_gap = unit(1, \"cm\"))\n",
    "dev.off()\n",
    "\n",
    "pdf(\"./results/figures/9_Dotplot_B_cell_cd3_pos_cd8_neg_CTLs_wt_bcl6_and_bcl6_double_cell1_time1_genotype1.pdf\", width = 12, height = 10)\n",
    "draw(hp2, annotation_legend_list = lgd, ht_gap = unit(1, \"cm\"))\n",
    "dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
