{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ab707625",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/liulab/galib/dlbcl_manuscript/scripts/10_infercnv'"
      ],
      "text/latex": [
       "'/liulab/galib/dlbcl\\_manuscript/scripts/10\\_infercnv'"
      ],
      "text/markdown": [
       "'/liulab/galib/dlbcl_manuscript/scripts/10_infercnv'"
      ],
      "text/plain": [
       "[1] \"/liulab/galib/dlbcl_manuscript/scripts/10_infercnv\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘tidyverse’ was built under R version 4.1.3”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.6      \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.8      \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.10\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0      \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.1 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2      \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1 \n",
      "\n",
      "Warning message:\n",
      "“package ‘tidyr’ was built under R version 4.1.2”\n",
      "Warning message:\n",
      "“package ‘readr’ was built under R version 4.1.2”\n",
      "Warning message:\n",
      "“package ‘forcats’ was built under R version 4.1.3”\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "Attaching sp\n",
      "\n",
      "here() starts at /liulab/galib/dlbcl_manuscript\n",
      "\n",
      "Warning message:\n",
      "“package ‘harmony’ was built under R version 4.1.3”\n",
      "Loading required package: Rcpp\n",
      "\n",
      "Warning message:\n",
      "“package ‘Rcpp’ was built under R version 4.1.2”\n",
      "Loading required package: viridisLite\n",
      "\n",
      "Warning message:\n",
      "“package ‘RColorBrewer’ was built under R version 4.1.3”\n",
      "Warning message:\n",
      "“package ‘Polychrome’ was built under R version 4.1.3”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'#FF00FF'</li><li>'#F400F4'</li><li>'#EA00EA'</li><li>'#DF00DF'</li><li>'#D500D5'</li><li>'#CA00CA'</li><li>'#BF00BF'</li><li>'#B500B5'</li><li>'#AA00AA'</li><li>'#9F009F'</li><li>'#950095'</li><li>'#8A008A'</li><li>'#800080'</li><li>'#750075'</li><li>'#6A006A'</li><li>'#600060'</li><li>'#550055'</li><li>'#4A004A'</li><li>'#400040'</li><li>'#350035'</li><li>'#2B002B'</li><li>'#200020'</li><li>'#150015'</li><li>'#0B000B'</li><li>'#000000'</li><li>'#000000'</li><li>'#0B0B00'</li><li>'#151500'</li><li>'#202000'</li><li>'#2B2B00'</li><li>'#353500'</li><li>'#404000'</li><li>'#4A4A00'</li><li>'#555500'</li><li>'#606000'</li><li>'#6A6A00'</li><li>'#757500'</li><li>'#808000'</li><li>'#8A8A00'</li><li>'#959500'</li><li>'#9F9F00'</li><li>'#AAAA00'</li><li>'#B5B500'</li><li>'#BFBF00'</li><li>'#CACA00'</li><li>'#D4D400'</li><li>'#DFDF00'</li><li>'#EAEA00'</li><li>'#F4F400'</li><li>'#FFFF00'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '\\#FF00FF'\n",
       "\\item '\\#F400F4'\n",
       "\\item '\\#EA00EA'\n",
       "\\item '\\#DF00DF'\n",
       "\\item '\\#D500D5'\n",
       "\\item '\\#CA00CA'\n",
       "\\item '\\#BF00BF'\n",
       "\\item '\\#B500B5'\n",
       "\\item '\\#AA00AA'\n",
       "\\item '\\#9F009F'\n",
       "\\item '\\#950095'\n",
       "\\item '\\#8A008A'\n",
       "\\item '\\#800080'\n",
       "\\item '\\#750075'\n",
       "\\item '\\#6A006A'\n",
       "\\item '\\#600060'\n",
       "\\item '\\#550055'\n",
       "\\item '\\#4A004A'\n",
       "\\item '\\#400040'\n",
       "\\item '\\#350035'\n",
       "\\item '\\#2B002B'\n",
       "\\item '\\#200020'\n",
       "\\item '\\#150015'\n",
       "\\item '\\#0B000B'\n",
       "\\item '\\#000000'\n",
       "\\item '\\#000000'\n",
       "\\item '\\#0B0B00'\n",
       "\\item '\\#151500'\n",
       "\\item '\\#202000'\n",
       "\\item '\\#2B2B00'\n",
       "\\item '\\#353500'\n",
       "\\item '\\#404000'\n",
       "\\item '\\#4A4A00'\n",
       "\\item '\\#555500'\n",
       "\\item '\\#606000'\n",
       "\\item '\\#6A6A00'\n",
       "\\item '\\#757500'\n",
       "\\item '\\#808000'\n",
       "\\item '\\#8A8A00'\n",
       "\\item '\\#959500'\n",
       "\\item '\\#9F9F00'\n",
       "\\item '\\#AAAA00'\n",
       "\\item '\\#B5B500'\n",
       "\\item '\\#BFBF00'\n",
       "\\item '\\#CACA00'\n",
       "\\item '\\#D4D400'\n",
       "\\item '\\#DFDF00'\n",
       "\\item '\\#EAEA00'\n",
       "\\item '\\#F4F400'\n",
       "\\item '\\#FFFF00'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '#FF00FF'\n",
       "2. '#F400F4'\n",
       "3. '#EA00EA'\n",
       "4. '#DF00DF'\n",
       "5. '#D500D5'\n",
       "6. '#CA00CA'\n",
       "7. '#BF00BF'\n",
       "8. '#B500B5'\n",
       "9. '#AA00AA'\n",
       "10. '#9F009F'\n",
       "11. '#950095'\n",
       "12. '#8A008A'\n",
       "13. '#800080'\n",
       "14. '#750075'\n",
       "15. '#6A006A'\n",
       "16. '#600060'\n",
       "17. '#550055'\n",
       "18. '#4A004A'\n",
       "19. '#400040'\n",
       "20. '#350035'\n",
       "21. '#2B002B'\n",
       "22. '#200020'\n",
       "23. '#150015'\n",
       "24. '#0B000B'\n",
       "25. '#000000'\n",
       "26. '#000000'\n",
       "27. '#0B0B00'\n",
       "28. '#151500'\n",
       "29. '#202000'\n",
       "30. '#2B2B00'\n",
       "31. '#353500'\n",
       "32. '#404000'\n",
       "33. '#4A4A00'\n",
       "34. '#555500'\n",
       "35. '#606000'\n",
       "36. '#6A6A00'\n",
       "37. '#757500'\n",
       "38. '#808000'\n",
       "39. '#8A8A00'\n",
       "40. '#959500'\n",
       "41. '#9F9F00'\n",
       "42. '#AAAA00'\n",
       "43. '#B5B500'\n",
       "44. '#BFBF00'\n",
       "45. '#CACA00'\n",
       "46. '#D4D400'\n",
       "47. '#DFDF00'\n",
       "48. '#EAEA00'\n",
       "49. '#F4F400'\n",
       "50. '#FFFF00'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"#FF00FF\" \"#F400F4\" \"#EA00EA\" \"#DF00DF\" \"#D500D5\" \"#CA00CA\" \"#BF00BF\"\n",
       " [8] \"#B500B5\" \"#AA00AA\" \"#9F009F\" \"#950095\" \"#8A008A\" \"#800080\" \"#750075\"\n",
       "[15] \"#6A006A\" \"#600060\" \"#550055\" \"#4A004A\" \"#400040\" \"#350035\" \"#2B002B\"\n",
       "[22] \"#200020\" \"#150015\" \"#0B000B\" \"#000000\" \"#000000\" \"#0B0B00\" \"#151500\"\n",
       "[29] \"#202000\" \"#2B2B00\" \"#353500\" \"#404000\" \"#4A4A00\" \"#555500\" \"#606000\"\n",
       "[36] \"#6A6A00\" \"#757500\" \"#808000\" \"#8A8A00\" \"#959500\" \"#9F9F00\" \"#AAAA00\"\n",
       "[43] \"#B5B500\" \"#BFBF00\" \"#CACA00\" \"#D4D400\" \"#DFDF00\" \"#EAEA00\" \"#F4F400\"\n",
       "[50] \"#FFFF00\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘ComplexHeatmap’ was built under R version 4.1.3”\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.10.0\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "  genomic data. Bioinformatics 2016.\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "Loading required package: usethis\n",
      "\n",
      "Warning message:\n",
      "“package ‘presto’ was built under R version 4.1.3”\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“package ‘ggpubr’ was built under R version 4.1.3”\n"
     ]
    }
   ],
   "source": [
    "getwd()\n",
    "setwd(\"/liulab/galib/dlbcl_manuscript/\")\n",
    "library(tidyverse)\n",
    "library(Seurat)\n",
    "library(here)\n",
    "library(harmony)\n",
    "library(viridis)\n",
    "library(RColorBrewer)\n",
    "library(Polychrome)\n",
    "PurpleAndYellow()\n",
    "library(ComplexHeatmap)\n",
    "library(devtools)\n",
    "library(presto)\n",
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(ggpubr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "849dc520",
   "metadata": {},
   "outputs": [],
   "source": [
    "B_cell<- readRDS(\"./data/objects/B_cell_final.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0558253b",
   "metadata": {},
   "outputs": [],
   "source": [
    "infercnv_pos_cells<- read.table('data/infercnv/10_B_cell_inferCNV_pos_PC5.tsv', header = FALSE, sep = '\\t')  %>% pull('V1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d1c1d43e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "42331"
      ],
      "text/latex": [
       "42331"
      ],
      "text/markdown": [
       "42331"
      ],
      "text/plain": [
       "[1] 42331"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(infercnv_pos_cells)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "86e68753",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_poolid_sick<- B_cell@meta.data  %>% filter(age == 'sick')  %>% \n",
    "    count(pool_id, sample_id, genotype, age)  %>% arrange(age, genotype)\n",
    "\n",
    "all_poolid_sick$genotype_new<- gsub(all_poolid_sick$genotype, pattern = '/[-+]', replacement = \"\")  %>% \n",
    "    gsub(pattern = ';', replacement = \"_\")  %>% gsub(pattern = '-', replacement = \"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f5997f08",
   "metadata": {},
   "outputs": [],
   "source": [
    "inpath = '/liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0efbe6ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_1_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_2_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_3_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_4_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_83_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_84_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_81_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_82_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_5_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_6_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_7_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_8_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_85_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_86_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_87_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n",
      "Reading /liulab/galib/mouse_scRNAseq_margaret/data/infercnv/inferCNV_out_new/Pool105_88_newsick/run.final.infercnv_obj ...\n",
      "\n",
      "'magick' package is suggested to install to give better rasterization.\n",
      "\n",
      "Set `ht_opt$message = FALSE` to turn off this message.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# with cluster annotation\n",
    "for (i in c(1:nrow(all_poolid_sick))){\n",
    "    \n",
    "    poolid = all_poolid_sick$pool_id[i]\n",
    "    genotype = all_poolid_sick$genotype[i]\n",
    "    age = all_poolid_sick$age[i]\n",
    "    sampleid = all_poolid_sick$sample_id[i]\n",
    "    \n",
    "    genotype_new<- gsub(genotype, pattern = '/[-+]', replacement = \"\")  %>% \n",
    "        gsub(pattern = ';', replacement = \"_\")  %>% gsub(pattern = '-', replacement = \"\")\n",
    "\n",
    "    message('Reading ', paste0(inpath,poolid, '_newsick/run.final.infercnv_obj ...'))\n",
    "    \n",
    "    infercnv_obj<- readRDS(paste0(inpath, poolid, '_newsick/run.final.infercnv_obj'))\n",
    "    gene_position = infercnv_obj@gene_order\n",
    "    mtx_obs<- infercnv_obj@expr.data[,infercnv_obj@observation_grouped_cell_indices[[genotype]]]\n",
    "    mtx_obs = t(mtx_obs)\n",
    "    mtx_ref<- infercnv_obj@expr.data[,infercnv_obj@reference_grouped_cell_indices[['WT']]]\n",
    "    mtx_ref = t(mtx_ref)\n",
    "    \n",
    "    # set heatmap row annotation\n",
    "    row_annotation = B_cell@meta.data  %>% filter(pool_id==poolid)  %>% select(new_annotation)  %>% mutate(infercnv = 'infercnv-neg')\n",
    "    row_annotation[rownames(row_annotation) %in% infercnv_pos_cells, 'infercnv'] = 'infercnv-pos'\n",
    "    row_annotation =  row_annotation %>% arrange(infercnv, new_annotation)\n",
    "\n",
    "    new_annotation_hex= B_cell@meta.data  %>% \n",
    "                                filter(pool_id==poolid)  %>% \n",
    "                                select(new_annotation, HEX_code)  %>% \n",
    "                                distinct()  %>% arrange(new_annotation)\n",
    "\n",
    "    new_annotation_colors = new_annotation_hex$HEX_code\n",
    "    names(new_annotation_colors) = new_annotation_hex$new_annotation\n",
    "\n",
    "    ra <- rowAnnotation(clusterID = row_annotation$new_annotation, infercnv = row_annotation$infercnv,\n",
    "                        col = list(infercnv = c(\"infercnv-pos\" = '#FF7F00', \"infercnv-neg\"= '#984EA3'),\n",
    "                                   clusterID = new_annotation_colors))\n",
    "\n",
    "    cnv_col_fun<- circlize::colorRamp2(c(1 - max(abs(mtx_obs - 1)), 1, 1 + max(abs(mtx_obs - 1))), c(\"blue\", \"white\", \"red\"))\n",
    "    cnv_col_fun<- circlize::colorRamp2(c(0.8, 1, 1.2), c(\"blue\", \"white\", \"red\"))\n",
    "    ht = Heatmap(mtx_obs[rownames(row_annotation),],\n",
    "            name = paste0(poolid, '_', sampleid), \n",
    "            border_gp = gpar(col = \"black\"),\n",
    "            column_split = factor(gene_position[colnames(mtx_obs), 'chr'], levels = unique(gene_position[['chr']])),\n",
    "            column_title_rot = 90,\n",
    "            column_title_gp = gpar(fontsize = 6),\n",
    "            column_gap = unit(0.1, \"mm\"),\n",
    "            cluster_rows = FALSE, \n",
    "            cluster_columns = FALSE, \n",
    "            show_row_names = FALSE,\n",
    "            show_column_names = FALSE, \n",
    "            right_annotation = ra,\n",
    "            use_raster = TRUE,\n",
    "            show_heatmap_legend = TRUE,\n",
    "            col = cnv_col_fun)\n",
    "    pdf(paste0('./results/figures/12_inferCNV_labeled_heatmap_', poolid, '_', sampleid, '_', genotype_new, '_', age, '_with_cluster.pdf'), width = 10, height = 6)\n",
    "    draw(ht)\n",
    "    dev.off()\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
